### PR TITLE
Fix typing.NamedTuple being not callable

### PR DESCRIPTION
**Pull request by rogalski**
Needed for PyCQA/pylint#1320

### CODE REVIEW

File: astroid/tests/unittest_inference.py [hunk 1]
```diff
@@ -3336,6 +3336,20 @@ def blurb(self):
...
+    @test_utils.require_version('3.5')
+    def test_typing_namedtuple_is_callable(self):
+        import typing
+        node = extract_node("""
+        import typing
+        MyNamedTuple = typing.NamedTuple('Named', [('foo', int), ('bar', int)])
+        MyNamedTuple
+        """)
+        inferred = list(node.infer())
+        if util.Uninferable in inferred:
```
Comment (rogalski) [inline astroid/tests/unittest_inference.py]: I don't like this early exit, but on the other hand `typing` is known to be actively developed, so I don't see much value in chasing issues related to older version of library code...

File: astroid/bases.py [hunk 2]
```diff
@@ -233,7 +233,7 @@ def __str__(self):
 
     def callable(self):
         try:
-            self._proxied.getattr('__call__', class_context=False)
+            self._proxied.getattr('__call__')
```
Comment (PCManticore) [inline astroid/bases.py]: What is the reason for this change?

File: astroid/tests/unittest_inference.py (hunk 1 above)
Comment (PCManticore) [inline astroid/tests/unittest_inference.py]: I don't like it either. Does this happen with older versions of ``typing``? We should, probably, pin it to a version that doesn't give us too much trouble.

File: astroid/bases.py (hunk 2 above)
Comment (rogalski) [inline astroid/bases.py]: typing.NamedTuple was inferred as not callable when class_context was False

Comment (PCManticore) [inline astroid/bases.py]: Yeah, but I'm asking from a protocol perspective. ``class_context`` is False, since the lookup is done on the instance, thus it should not go through the metaclass lookup and so on. What I'm asking is why typing.NamedTuple is special. Is it an instance or a class? If it is an Instance, how come it has a ``__call__`` defined in the metaclass (I presume), that it can access in this place? Where does its ``__call__`` come from?

**Comment by rogalski** [comment]
I'll close it - it's much better solved by #390
